#DPATCHLEVEL=1

2006-01-15  Samuel Thibault  <samuel.thibault@ens-lyon.org>

	Fix de4x5 PCI probe.

	* linux/src/drivers/net/de4x5.c (pci_probe): Keep probing only
	while pcibios_find_class succeeds.

--- gnumach-20050801/linux/src/drivers/net/de4x5.c	1999-04-26 07:52:00.000000000 +0200
+++ gnumach-mine/linux/src/drivers/net/de4x5.c	2006-01-15 00:35:45.000000000 +0100
@@ -2147,7 +2147,7 @@ pci_probe(struct device *dev, u_long ioa
     }
 
     for (index=lastPCI+1; 
-	 (pcibios_find_class(class, index, &pb, &dev_fn)!= PCIBIOS_DEVICE_NOT_FOUND);
+	 (pcibios_find_class(class, index, &pb, &dev_fn)== PCIBIOS_SUCCESSFUL);
 	 index++) {
 	dev_num = PCI_SLOT(dev_fn);
 	if ((!pbus && !dnum) || ((pbus == pb) && (dnum == dev_num))) {

